### DISCUSSION TITLE
Hidden categories - how?

### DISCUSSION THREAD

**Discussion by giannello** [Smile-SA/elasticsuite#2621] [Q&A]
Hello there!

According to the [Magento 2 documentation](https://docs.magento.com/user-guide/catalog/category-hidden.html), it's possible to create a "hidden" category, that is not shown anywhere, but can be linked and used if wanted.

When using elasticsuite, this option is not present anymore (because of [this](https://github.com/Smile-SA/elasticsuite/blob/9ad6ffaac8eb2ded40366fe3b5374484208f0f4c/src/module-elasticsuite-catalog/Setup/CatalogSetup.php#L98)).

Why does elasticsuite disable this magento feature, and how can I achieve the same result if that option is not available anymore?

**Comment by rbayet** [comment]
Hello @giannello,

It is a legacy design decision to ease the way the category layer navigation filter and the category related rule engine works.
This prevents having to iterate on the children categories of a given category to determine if their product are supposed to be show in that category too and having to deal with the fact of a child category which would **not** be anchor, but have itself anchor children categories.
Should the grand-children categories products be show in the original category ?
That sort of things.

**But** all is not lost if you want somehow to hide (a bit) a given category : 
* you can still use the "Include in Menu = No", so it will not be displayed in your top menu (it will still be show in your layered navigation menu, though)
* you can prevent the category from being displayed in the categories autocomplete search results by switching off 
![image](https://user-images.githubusercontent.com/2353374/183829463-300ab8c1-a7d0-4dad-a9e4-8fab28f04f7c.png)
* even though its use is intended to avoid different kind of products to be found when the category name is a composite (eg "Wood panels, floor tiles, stone tiles") and searching for a single concept (eg "wood panel"), you can even prevent searching for that category name to return children and grand-chidren products by switching off 
![image](https://user-images.githubusercontent.com/2353374/183830179-281dbbba-0579-4e51-99af-b1fa47b01c9e.png)

If you really need to also hide the category in the layered navigation, you could rewrite the category layered navigation filter by making it skip children categories of the current category flagged with a custom category attribute (or is_anchor = false), see https://github.com/Smile-SA/elasticsuite/blob/1a42d60a56bbb80724f2a02f5cf92103eeceb3d6/src/module-elasticsuite-catalog/Model/Layer/Filter/Category.php#L144-L155

And if you do not want to create a custom category attribute, you can make is_anchor available again in the admin category edit form by redeclaring the field `<field name="is_anchor">` as it is declared by default in Magento legacy in custom UI components layout overrides, to counter the effects of 
* https://github.com/Smile-SA/elasticsuite/blob/e8e67b407f3f49a2bac16e3525173d19af9c0f7c/src/module-elasticsuite-catalog/view/adminhtml/ui_component/catalogstaging_category_update_form.xml#L24-L31
* https://github.com/Smile-SA/elasticsuite/blob/9ef1b16b708f099d43388868b9eab40dee1fb2d4/src/module-elasticsuite-catalog/view/adminhtml/ui_component/category_form.xml#L24-L31

But we have never tried this and you will be a bit on your own ;)

Regards,

**Comment by giannello** [comment]
Hi @rbayet, and thanks for the detailed explanation.

I tried re-enabling that field on my local environment (directly by changing the attribute visibility in the database) but ultimately abandoned the solution due to the potential of breakages.

As it's possible to hide categories in the navigation menu, and our shop doesn't allow for a _global_ product listing, as we always start with a sub-category, it sounds like it might be worth creating those hidden categories as top-level categories, or as sub-categories of a non-visible first-level category, to prevent leaking into the layered navigation. Will give it a spin
